hack/release.sh

```diff
@@ -23,23 +23,15 @@ readonly EVENTING_RELEASE_GCS
 readonly EVENTING_RELEASE_GCR
 
 # Yaml files to generate, and the source config dir for them.
+declare -A COMPONENTS
+COMPONENTS["eventing.yaml"]="config"
+COMPONENTS["in-memory-channel.yaml"]="config/provisioners/in-memory-channel"
```

Comment:
Won't this duplicate the config that's already in `eventing.yaml`, since that file contains this directory also? Then the release yaml will have two copies of the in-memory-channel objects.

Reply:
@grantr There will be three files produces for the release:
- `eventing.yaml` only knative eventing
- `in-memory-channel.yaml` only the in-memory provisioner
- `release.yaml` eventing.yaml + in-memory-channel.yaml

When you [give a directory to ko](https://github.com/knative/eventing/pull/532/files/7aa9f7582369cad12e317c9d07e68910b771a254#diff-cede583a27c227d458ca67079957c8a2R61), it will only build config files that are in that exact directory.

I ran this script with my own creds when I originally put up the PR.

Reply:
:+1: Thanks, TIL. Apparently there's a `--recursive` flag to `kubectl` that selects the behavior I was expecting.

Minor preference for `core.yaml` or `eventing-core.yaml` over `eventing.yaml`, but this is fine for now.

/hold cancel

Reply:
This is the patten that serving uses, but it's easy enough to change.